File: backend/src/main/java/com/quantix/backend/dto/AuthResponse.java
package com.quantix.backend.dto;

import com.quantix.backend.entity.User;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private String token;
    private String tokenType;
    private AuthUser user;

    public AuthResponse(String token, User user) {
        this.token = token;
        this.tokenType = "Bearer";
        this.user = new AuthUser(user);
    }
}

File: mobile/app/src/main/java/com/quantix/mobile/data/model/response/AuthResponse.kt
package com.quantix.mobile.data.model.response

data class AuthResponse(
    val token: String,
    val tokenType: String,
    val user: UserData
)

data class UserData(
    val id: Long,
    val email: String,
    val firstName: String,
    val lastName: String,
    val role: Role
)

data class Role(
    val roleId: Long,
    val roleName: String
)